spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingProperties.java

```diff
@@ -84,8 +113,120 @@ else if (binding instanceof Map) {
 		return channelName;
 	}
 
+	public int getPartitionCount(String channelName) {
+		Object binding = bindings.get(channelName);
+		// we may shortcut directly to the path
+		if (binding instanceof Map) {
+			try {
+				Map<?, ?> bindingProperties = (Map<?, ?>) binding;
+				Object bindingPath = bindingProperties.get(PARTITION_COUNT);
+				if (bindingPath != null) {
+					return Integer.parseInt(bindingPath.toString());
+				}
+			} catch (NumberFormatException e) {
+				// ignore and just return 1
+			}
+		}
+		return 1;
+	}
+
+	public boolean isPartitionedConsumer(String channelName) {
+		Object binding = bindings.get(channelName);
+		// if the setting is just a target shortcut
+		if (binding == null || binding instanceof String) {
+			return false;
+		}
+		else if (binding instanceof Map) {
+			Map<?, ?> bindingProperties = (Map<?, ?>) binding;
+			Object bindingPath = bindingProperties.get(PARTITIONED);
```

Comment:
ok, I thought default way of `partitioning` would help distributing the load. But I see it can't help always. Competing consumers would help in some cases.